### PR TITLE
Implement NextJS to Scan Reports Value view

### LINKED ISSUE #677: Fields with large number of values are not loading
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Currently the large scan reports are uploading with no errors, however fields with large number of values show 'loading values' status and then the page stops responding. 

### Expected Behavior

Show all the field values 

### Steps To Reproduce

Scan report # 620
Table : share_gp_data.csv
Field: NewRead

### Environment

```markdown
- OS:
- Other environment details:
```

### I'm part of a Project Team

_No response_

### Anything else?

_No response_

### Are you willing to contribute to resolve this issue?

None

### LINKED ISSUE #681: Scan Report Values
# Summary

Rewrite this view with Nextjs.

Example: `/scanreports/226/tables/840/fields/12970/`

Frontend Tasks:

- [ ] Recreate the existing UI
- [ ] Define types in Typescript for what is needed
- [ ] Add pagination
- [ ] Add filtering
- [ ] Add sorting
- [ ] Add the "Add Concept" POST
- [ ] Review

Backend Tasks:

- [ ] New API endpoint
- [ ] With pagination, filtering, and sorting
- [ ] Testing 
- [ ] Add the URLS to proxy it 
- [ ] Review

### PR DESCRIPTION
**Pull request by AndrewThien**
# Changes

This PR implemented NextJS to build the page Values of the selected Scan report. 

- Server-side sorting, pagination and filtering were applied
- Add Concept function has the validations of input Concept ID placed in Django Backend  

Closes #681 
Closes #677 

# Screenshots

Screenshot 2024-05-28 at 11 47 10

# Checks

**Important:** please complete these **before** merging.
- [x] Run migrations, if any.
- [x] Update `changelog.md`, including migration instructions if any.
- [x] Run unit tests.

### CODE REVIEW

**Comment by spco** [comment]
Is it possible to set the default number of rows to a larger number on this view? 10 per page will almost always be annoyingly small, when these fields can contain >1000 values. 25 might be a more suitable default and still give good performance. @erummas or @armando-mv might be able to weigh in with a default value? (I know the user can change this at the bottom of the page, but the users will prefer not to have to do that every time, I'm sure).

File: app/next-client-app/components/concepts/add-concept.tsx [hunk 1]
```diff
@@ -31,12 +37,17 @@ export default function AddConcept({ id, tableId }: AddConceptProps) {
...
-        object_id: id,
-        content_type: "scanreportfield",
+        object_id: rowId,
+        content_type:
+          location === "SR-Values" ? "scanreportvalue" : "scanreportfield",
         creation_type: "M",
-        table_id: tableId,
+        table_id: fieldObj ? fieldObj.scan_report_table : parentId,
       });
       toast.success("ConceptId linked to the value");
```
[the remark this answers is not in the document]
Comment (AndrewThien) [reply app/next-client-app/components/concepts/add-concept.tsx]: Hi @spco , the successful adding Concept message has been updated

**Comment by AndrewThien** [comment]
> Is it possible to set the default number of rows to a larger number on this view? 10 per page will almost always be annoyingly small, when these fields can contain >1000 values. 25 might be a more suitable default and still give good performance. @erummas or @armando-mv might be able to weigh in with a default value? (I know the user can change this at the bottom of the page, but the users will prefer not to have to do that every time, I'm sure).

About this one, I have changed the value to 25, as requested, but was not able to test it on my end (because I don't have the test data having more than 25 values in one field). So can you please check on your end to see if the change is well implemented, @spco ? Thanks

File: app/next-client-app/components/concepts/add-concept.tsx [hunk 2]
```diff
@@ -31,14 +37,19 @@ export default function AddConcept({ id, tableId }: AddConceptProps) {
 
   const handleSubmit = async (conceptCode: number) => {
     try {
+      let fieldObj;
```
Comment (AndyRae) [inline app/next-client-app/components/concepts/add-concept.tsx:40]: This is a little confusing to read, wondering if it'd be easier something like

```
const contentType = location === "SR-Values" ? "scanreportvalue" : "scanreportfield";

const tableId = location === "SR-Values" 
    ? (await getScanReportField(parentId)).scan_report_table 
    : parentId;
```
I'd almost be tempted to extract this logic to a method, so the logic is encapsulated and cleaner. like `determineContentType` or something.

**Review by AndyRae — CHANGES_REQUESTED** [review]

**Comment by AndyRae** [comment]
> About this one, I have changed the value to 25, as requested, but was not able to test it on my end (because I don't have the test data having more than 25 values in one field). So can you please check on your end to see if the change is well implemented, @spco ? Thanks

This change works for me locally - I think 25 is a nice default for now.

**Comment by AndyRae** [comment]
It would be nice to preserve the tooltips also, for example:

Screenshot 2024-05-28 at 12 37 14

The reason being - they give a readable example of what the (V/M/R) actually mean.

**Review by AndyRae — CHANGES_REQUESTED** [review]

**Comment by AndrewThien** [comment]
I also modified the pagination buttons' styling, making them neater a bit. Please re-review, @AndyRae

**Review by AndyRae — APPROVED** [review]

**Review by dcl10 — APPROVED** [review]

**Review by Jivanam — APPROVED** [review]

**Pull request merged**